This pull request includes the changes for InfluxDB Logger to enable output to InfluxDB2.x

Mavrrick opened the pull request:
The first set of changes start out by modifying the ui of the app. This is needed as we need to enable the ability to select what version of Influxdb a user is using and then in turn what version of authentication a user will use. To do this we first declare two pages for the setup. The initial page which used a non-descriptive title is was updated to "setupMain" and a second new page labeled "connectionPage"

The "setupMain" page was updated to remove the configuration options for the database and had a link added to load the "connectionPage"

The "connectionPage" allows for selecting between the various options needed between InfluxDB1.x and InfluxDB2.x

The second group of updates where to the update() method Database config section. This section was update to account for the configuration values added to handle InfluxDB2.x. Along with that there was code added to handle building the URL string and header packages needed depending on what version of influxdb the user is running.

The last update was to the postToInfluxDb method to handle the difference that may occur between influxdb versions. The code that was in place for the URI value for Influxdb1.x was hard coded to much and not usable with 2.x. It was updated to use a URI variable that is built in the update method mentioned above.

Mavrrick committed:
Update influxdb-logger.groovy

Mavrrick committed:
Update influxdb-logger.groovy

Mavrrick committed:
default to basic auth type

Mavrrick committed:
Update influxdb-logger.groovy

Mavrrick commented:
There is one catch to all of this. Because the Database values and connection strings are a bit different between v1 and v2  a user will need to open InfluxDB after the updated confirm they have they correct authorization type, and click on done so that it can run the update method. I set the values to default to what Influxdb1.x would need, but they still need to atleast trigger the update method to run for the app to ensure everything new is created after the code deploy. I don't know of any way around this need when adding this kind of functionality.

Mavrrick committed:
Update version for HPM

Update version # for HPM and add notes about update

Mavrrick committed:
added to modification history

Added comments about updates

dennypage reviewed (changes requested):
A few things...

- The changes revert some of the indentation and tab vs space fixes. Please stick to spaces and the existing indentation level.
- Please stick to the existing coding style. Lint driven.
- You have a commented out block of code, similar to the blocks you removed in the prior PR. Is this intended?
- Since you are creating state.uri, is there a reason to create state.databaseHost, state.databasePort, state.databaseTls, state.databaseName and state.bucket?

 Thanks.

Mavrrick committed:
Removed Commented section of code

Mavrrick commented:
1. Yea. I apologize for that. I keep trying different things and it looks ok one second and then not right when i got back to it. What settings are you using for the spacing. maybe I have something out of sync with what you are using. In the editor it is showing i am using a tab with spacing of 4 when things look right.
2. What does Lint Driven mean. What I have researched looks like it just means i put it through a code scanner.
3. It was intended as I tend to not remove code on my first run through incase we find a reason to revert it later. I have removed the commented code in my last update to the file. Not really needed when doing this in github and we can look at the history.
4. Those steps were pulled from the code I found that added support for InfluxDBv2.x. I wasn't modifing it, but just coping/pasteing the code so it was known good code and could be included. When I reviewed the code that uses all of those values it appears they are all about creating the correct url string in state.uri. The person that originally wrote used all of those values to build the correct URL string. I can look at changing the method to create the URL string, but this code is working

Mavrrick committed:
Attempt to get spacing corrected

Mavrrick committed:
Corrected spacing a second time using spaces only

dennypage commented:
No problem. Lint is a tool that is used to find simple issues in code and to also to unify code formatting. I am using VSCodium with the Groovy lint plugin. Note that Groovy lint will complain about a lot of things--I've disabled a bunch of checks with the first line in the file. It was just too much.

Mavrrick commented:
I did further digging into the state values and identified that they don't really need to be included.  It does look like i can make some changes to remove some of those state values created. I will publish another update with these change and the state values removed from program. I also found a single logging line that will need to be updated for this removal of the state values identified.

Mavrrick committed:
Remove State values and updated URL creation process

Removed State values that are only used to build the State.uri value
Update process that creates the state.uri value to use settings values instead of state values
Updated postToInfluxDB method with new logger code to not use removed state values.

dennypage commented:
FYI, here's a (bad cut/paste) sample of the current lint output:

The map entry hidden:false within class None is not preceded by a space or whitespace
The map entry hidden: false within class None is not preceded by a space or whitespace
The operator "-" within class None is not preceded by a space or whitespace
The operator "=" within class None is not followed by a space or whitespace
The if keyword within class None is not followed by a single space
The if keyword within class None is not followed by a single space
The operator "-" within class None is not followed by a space or whitespace
The closure arrow (->) within class None is not surrounded by a space or whitespace
The opening brace for the closure in class None is not followed by a space or whitespace
The operator "+" within class None is not preceded by a space or whitespace
The operator "+ " within class None is not followed by a space or whitespace
The operator "+" within class None is not preceded b a space or whitespace
The operator "+" within class None is not followed by a space or whitespace
The operator "," within class None is not surrounded by a space or whitespace
The operator "?" within class None is not surrounded by a space or whitespace
There is whitespace between the method name and parenthesis in a method call: input
There is whitespace between the method name and parenthesis in a method call: input
Line ends with whitespace characters
Line ends with whitespace characters
Semicolons as line endings can be removed safely
Semicolons as line endings can be removed safely
Semicolons as line endings can be removed safely
Semicolons as line endings can be removed safely
Line ends with whitespace characters
The operator "+" within class None is not preceded by a space or whitespace
The operator "+" within class None is not followed by a space or whitespace
Semicolons as line endings can be removed safely
Line ends with whitespace characters
Line ends with whitespace characters
Line ends with whitespace characters
t.groovy [Ln 102, Col 42] t.groovy [Ln 145, Col 421 t.groovy [Ln 148, Col 31] t.groovy [Ln 148, Col 31] t.groovy [Ln 150, Col 171 t.groovy [Ln 152, Col 21] t.groovy [Ln 153, Col 21] t.groovy [Ln 154, Col 391 t.groovy [Ln 156, Col 25] t.groovy [Ln 156, Col 35] t.groovy [Ln 158, Col 42] t.groovy [Ln 158, Col 42] t.groovy [Ln 163, Col 77] t.groovy [Ln 163, Col 77] t.groovy [Ln 176, Col 521 t.groovy [Ln 176, Col 91] t.groovy [Ln 177, Col 18] t.groovy [Ln 194, Col 18] t.groovy [Ln 220, Col 9] t.groovy [Ln 280, Col 45] t.groovy [Ln 284, Col 11 t.groovy [Ln 290, Col 19] t.groovy [Ln 292, Col 32] t.groovy [Ln 294, Col 31] t.groovy [Ln 296, Col 36] t.groovy [Ln 297, Col 1] t.groovy [Ln 299, Col 19] t.groovy [Ln 299, Col 19] t.groovy [Ln 299, Col 44] t.groovy [Ln 301, Col 1] t.groovy [Ln 308, Col 24] t.groovy [Ln 309. Col 1]

While I'm looking at the code, would you mind fixing stuff like this up? Easiest way would be using a tool which can fix almost all of it automatically. Thanks.

Mavrrick committed:
Code after being scanned by lint tool

Mavrrick commented:
I got VSCodium downloaded and I believe I managed to scan the code and get everything fixed that came up. At this point i think all of your concerns should be addressed. All of the spacing should be better, the code has been scanned by a lint analyzer, The comments that were added have been removed.  Code was updated to remove unnecessary state values and updated to build the URL string from the Settings values instead of from state values only created for that purpose. There was also a logging line updated to account for some of the state values being removed and not be avaliable for the logging function.

Mavrrick committed:
Corrected missing configuration value that was removed during cleanup.

It appears during one of the attempts to cleanup the code the perfORG settings value was removed. Added back in

dennypage commented:
Okay, thanks. I will take another look. I want to do a little testing prior to merge.
Have you tested this version with Influx 1.x and/or Influx 2.x?

dennypage commented:
One last set of lint fixes--diff below.

Also, can you temporarily pull the change to PackageManifest.json please? I will have a follow-on commit, which I will ask you to review and test, before this gets pushed out to everyone.

Thanks!

----------
[Edit: forgot to protect the diff!]


```
61c62
<             description: prefDatabaseHost==null?"Configure database connection parameters":prefDatabaseHost,
---
>             description : prefDatabaseHost == null ? "Configure database connection parameters" : prefDatabaseHost,
66c67
<             input (
---
>             input(
101c102
<         if(!accessAllAttributes) {
---
>         if (!accessAllAttributes) {
148c149
<             state.selectedAttr=[:]
---
>             state.selectedAttr = [:]
150c151
<                 if(deviceName) {
---
>                 if (deviceName) {
153,154c154,155
<                     if(attr) {
<                         state.options =[]
---
>                     if (attr) {
>                         state.options = []
156c157
<                         attr.each {at->
---
>                         attr.each { at->
158c159
<                             index = index+1
---
>                             index = index + 1
163c164
<                         state.selectedAttr[deviceId] = settings["attrForDev"+deviceId]
---
>                         state.selectedAttr[deviceId] = settings["attrForDev" + deviceId]
176,177c177,178
<             input "prefDatabasePort", "text", title: "Port", defaultValue: prefDatabaseTls?"443":"8086", required: false
<             input (
---
>             input "prefDatabasePort", "text", title : "Port", defaultValue : prefDatabaseTls ? "443" : "8086", required : false
>             input(
195c196
<             input (
---
>             input(
280c281
<     state.uri = "";
---
>     state.uri = ""
282c283
<         state.uri += "https://";
---
>         state.uri += "https://"
284c285
<         state.uri += "http://";
---
>         state.uri += "http://"
286c287
<     state.uri += settings.prefDatabaseHost;
---
>     state.uri += settings.prefDatabaseHost
289c290
<         state.uri += ":"+settings.prefDatabasePort;
---
>         state.uri += ":" + settings.prefDatabasePort
748c749
<         //if(!mutex.tryAcquire()) {
---
>         //if (!mutex.tryAcquire()) {
777c778
<         //if(!mutex.tryAcquire()) {
---
>         //if (!mutex.tryAcquire()) {

```

Mavrrick committed:
Update PackageManifest.json

Mavrrick committed:
Updated with latest lint driven updates requested

Mavrrick commented:
I have reverted back the PackageManifest.json file so it won't prompt updates from HPM. I have also gone through and made the requested changes based on your scan with the lint analyzer.

I have been testing this on my dev hub with both versions of InfluxDB and it has been working. The only thing I haven't been able to test myself is the SSL flag as i think that is for users with INfluxDB Cloud accounts. I have also reinstalled it multiple times and tried to upgrade from old code to the new code a few times as well. So it looks good, but i am also anxious to get more eyes on it to see if there are any problems. 

Any time i have seen errors has been 1. I upgraded and didn't validate the connection configuration and click on done from the app after upgrading. 2. When I have missed a setting in the configuration like an incorrect ip or port or wrong authentication for Influxdb1 or 2  based on the database i was hitting. I have actually put the code in my repo on my production hub now so I am already running the code.

I do have one more change after this to add a configuration value for the size of the queue before forcing a update. I have found reducing the number to 75 instead of 100 that is hard coded helps reduce load in my instance. That will be the last update i can think of once this is applied

dennypage commented:
Okay, I'm good with the base changes. My additional changes can be seen pr #29
